client/src/pages/Compare.js

```diff
@@ -0,0 +1,49 @@
+import React, { useEffect, useState } from "react";
+import { fetchSupermarkets } from "../api/supermarkets";
+import Header from "../components/Header";
+import useAsync from "../hooks/useAsync";
+import PropTypes from "prop-types";
+import SupermarketDisplay from "../components/SupermarketDisplay";
+import getCompareListsData from "../components/getCompareListsData";
+
+function Compare({ listData }) {
+  const { data: supermarkets, loading, error } = useAsync(fetchSupermarkets);
+  const [totalPrices, setTotalPrices] = useState("");
+  const [connectionFail, setConnectionFail] = useState(true);
+
+  useEffect(() => {
+    getCompareListsData({
+      supermarkets,
+      listData,
+      setConnectionFail,
+      setTotalPrices,
+    });
+  }, [supermarkets, listData]);
+  return (
+    <>
+      <Header title={"Compare"} />
+      {connectionFail && (
+        <div>
+          Your session has ended. Please navigate back to your grocery list
+        </div>
+      )}
+      {error && <div>Could not get data. Dont cry. Try again</div>}
```

Comment:
Since your `fetchSupermarkets` throws an error you could have the UI show the error message here :)

Reply:
Hey, thanks for your feedback. I opend this branch from my other PR that´s why all the commits and changes are present. The commits with the green hook are the ones I changed here :)

Reply:
Gotcha :)